### PR TITLE
Enable Typescript codesharing between OperatorUI & Styleguide

### PR DESCRIPTION
**Pull request by rupurt**
- Enforce type checking
- Fixes type checking errors
- Build artifacts must be checked into the repo `yarn workspace @chainlink/styleguide run build`

### CODE REVIEW

File: styleguide/src/components/Cards/SimpleList.jsx.map [hunk 1]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"SimpleList.jsx","sourceRoot":"","sources":["SimpleList.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAO,KAAK,MAAM,yBAAyB,CAAA;AAC3C,OAAO,SAAS,MAAM,6BAA6B,CAAA;AACnD,OAAO,IAAI,MAAM,wBAAwB,CAAA;AACzC,OAAO,SAAS,MAAM,SAAS,CAAA;AAO/B,MAAM,UAAU,GAAG,CAAC,EAAE,QAAQ,EAAE,KAAK,EAAU,EAAE,EAAE;IACjD,OAAO,CACL,CAAC,IAAI,CACH;MAAA,CAAC,SAAS,CAAC,CAAC,KAAK,CAAC,EAAE,SAAS,CAE7B;;MAAA,CAAC,KAAK,CAAC,KAAK,CAAC,CAAC,EAAE,WAAW,EAAE,OAAO,EAAE,CAAC,CACrC;QAAA,CAAC,SAAS,CAAC,CAAC,QAAQ,CAAC,EAAE,SAAS,CAClC;MAAA,EAAE,KAAK,CACT;IAAA,EAAE,IAAI,CAAC,CACR,CAAA;AACH,CAAC,CAAA;AAED,eAAe,UAAU,CAAA"}
```
Comment (se3000) [inline styleguide/src/components/Cards/SimpleList.jsx.map]: Should we be checking in `.map` files? This is [the only opinion I could find](https://stackoverflow.com/questions/26462852/typescript-which-files-should-i-check-in) after a quick search, and it doesn't seem very definitive. Also, I see this is in `src` and I'm wondering checking it in is actually necessary.

Comment (rupurt) [reply styleguide/src/components/Cards/SimpleList.jsx.map]: I feel like we should commit them because it helps with navigation. Otherwise each dev would need to run `yarn workspace @chainlink/styleguide build` when they pull fresh changes.

Comment (NavyAdmiral) [reply styleguide/src/components/Cards/SimpleList.jsx.map]: sorry for interjecting but hook it into a postinstall? 
ill take a review look tonight

**Comment by rupurt** [comment]
@se3000 @NavyAdmiral I had to update the base branch for this which dismissed the review

File: operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-import Grid from '@material-ui/core/Grid'
-import StatusItem from 'components/JobRuns/StatusItem'
-import capitalize from 'lodash/capitalize'
-import { IInitiator, ITaskRuns, IJobRun } from '../../../@types/operator_ui'
 import { createStyles } from '@material-ui/core'
 import { withStyles, WithStyles } from '@material-ui/core/styles'
+import Grid from '@material-ui/core/Grid'
+import capitalize from 'lodash/capitalize'
+import {
+  IInitiator,
```
Comment (NavyAdmiral) [inline operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx]: this and below are unused

File: operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx [hunk 3]
```diff
@@ -22,100 +27,137 @@ const fontStyles = () =>
...
 
 interface IItemProps extends WithStyles<typeof fontStyles> {
-  keyOne: string
-  valOne: string
-  keyTwo: string
-  valTwo: string
+  colATitle: string
+  colAValue: string
+  colBTitle: string
+  colBValue?: string | null
```
Comment (NavyAdmiral) [inline operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx]: good change, but isnt `?: | null` redundant?

File: operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx [hunk 4]
```diff
@@ -22,100 +27,137 @@ const fontStyles = () =>
...
 
 const Item = withStyles(fontStyles)(
-  ({ keyOne, valOne, keyTwo, valTwo, classes }: IItemProps) => (
+  ({ colATitle, colAValue, colBTitle, colBValue, classes }: IItemProps) => (
     <Grid container>
       <Grid item sm={2}>
-        <p className={classes.header}>{keyOne}</p>
-        <p className={classes.subHeader}>{valOne || 'No Value Available'}</p>
+        <p className={classes.header}>{colATitle}</p>
+        <p className={classes.subHeader}>{colAValue}</p>
```
Comment (NavyAdmiral) [inline operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx]: dont remember exactly what's going on here conceptually, but why isn't this `nullable` as the other col?

File: operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx [hunk 5]
```diff
@@ -22,100 +27,137 @@ const fontStyles = () =>
...
+                  {taskRun.task && <Params params={taskRun.task.params} />}
+                </Grid>
+                <Grid item>
+                  <Result run={taskRun} />
+                </Grid>
+              </Grid>
+            </StatusItem>
+          </Grid>
+        )
+      })}
```
Comment (NavyAdmiral) [inline operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx]: overall good changes here too, but the changes in this file might deserve a separate PR

File: operator_ui/src/components/Link.tsx [hunk 6]
```diff
@@ -27,7 +27,7 @@ const styles = (_theme: Theme) =>
   })
 
 interface IProps extends WithStyles<typeof styles> {
-  children: React.ReactNode
+  children: React.ReactNode | string
```
Comment (NavyAdmiral) [inline operator_ui/src/components/Link.tsx]: isn't `| string` redundant since `React.ReactNode` covers string (i could be wrong)?

File: operator_ui/src/connectors/redux/reducers/jobRuns.ts [hunk 7]
```diff
@@ -1,37 +1,54 @@
 import { RECEIVE_DELETE_SUCCESS } from '../../../actions'
```
Comment (NavyAdmiral) [inline operator_ui/src/connectors/redux/reducers/jobRuns.ts]: ✂️

File: operator_ui/src/containers/JobRuns/Show/ErrorLog.tsx [hunk 8]
```diff
@@ -9,17 +9,19 @@ import {
 } from '@material-ui/core/styles'
 import Grid from '@material-ui/core/Grid'
 import Typography from '@material-ui/core/Typography'
+import PaddedCard from '@chainlink/styleguide/src/components/PaddedCard'
 import { fetchJobRun } from '../../../actions'
 import jobRunSelector from '../../../selectors/jobRun'
-import PaddedCard from '../../../components/PaddedCard'
 import PrettyJson from '../../../components/PrettyJson'
```
Comment (NavyAdmiral) [inline operator_ui/src/containers/JobRuns/Show/ErrorLog.tsx]: ✂️ this and `createStyles` in line 5

File: operator_ui/src/selectors/jobsShowRunCount.ts [hunk 9]
```diff
@@ -1,3 +1,3 @@
-export default state => {
-  return state.jobRuns.currentJobRunsCount
-}
+import { IState } from '../connectors/redux/reducers/index'
+
+export default ({ jobRuns }: IState) => jobRuns.currentJobRunsCount
```
Comment (NavyAdmiral) [inline operator_ui/src/selectors/jobsShowRunCount.ts]: lets add return type too

Comment (rupurt) [reply operator_ui/src/selectors/jobsShowRunCount.ts]: This doesn't need to return type as it's inferred

File: operator_ui/src/components/JobRuns/StatusCard.tsx [hunk 10]
```diff
@@ -7,14 +7,14 @@ import {
...
-import PaddedCard from '@chainlink/styleguide/components/PaddedCard'
+import PaddedCard from '@chainlink/styleguide/src/components/PaddedCard'
 import { titleCase } from 'change-case'
 import StatusIcon from '../JobRuns/StatusIcon'
 import { Grid } from '@material-ui/core'
 import { IJobRun } from '../../../@types/operator_ui'
 import ElapsedTime from '../ElapsedTime'
 
-const styles = (theme: Theme) =>
+const styles = (theme: any) =>
```
[the remark this answers is not in the document]
Comment (rupurt) [reply operator_ui/src/components/JobRuns/StatusCard.tsx]: Yeah, custom fields. I've left the `Theme` type for ones that don't have custom fields

File: operator_ui/src/components/Cards/TokenBalance.js [hunk 11]
```diff
@@ -3,8 +3,8 @@ import PropTypes from 'prop-types'
 import numeral from 'numeral'
```
[the remark this answers is not in the document]
Comment (rupurt) [reply operator_ui/src/components/Cards/TokenBalance.js]: This is a `.js` file which is probably why it hasn't complained yet. When it's converted to `.tsx` it will need to bring in the types.

File: operator_ui/src/connectors/redux/reducers/jobRuns.ts [hunk 12]
```diff
@@ -1,37 +1,54 @@
...
+export type Action =
+  | { type: 'UPSERT_JOB_RUNS'; data: any }
+  | { type: 'UPSERT_RECENT_JOB_RUNS'; data: any }
+  | { type: 'UPSERT_JOB_RUN'; data: any }
+  | { type: 'RECEIVE_DELETE_SUCCESS'; response: any }
+
+export default (state: IState = initialState, action: Action) => {
   switch (action.type) {
-    case UPSERT_JOB_RUNS: {
+    case 'UPSERT_JOB_RUNS': {
```
Comment (NavyAdmiral) [inline operator_ui/src/connectors/redux/reducers/jobRuns.ts]: will using the consts here from lines 16-18 cause type errors with `Action`? those exports are used only in 1 file and other files use string literals, so maybe its better to deprecate the const vars fpor action names entirely?

Comment (rupurt) [reply operator_ui/src/connectors/redux/reducers/jobRuns.ts]: Removed usage of `const`s in the tests

File: operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx (hunk 3 above)
Comment (NavyAdmiral) [reply operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx]: @rupurt reminder about this

File: operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx (hunk 4 above)
Comment (NavyAdmiral) [reply operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx]: @rupurt reminder about this
